src/priceClient/priceClient.e2e.ts

```diff
@@ -0,0 +1,153 @@
+import assert from "assert";
+import dotenv from "dotenv";
+import winston from "winston";
+import { Logger, msToS, PriceCache, PriceClient, TokenPrice } from "./priceClient";
+import { across, coingecko } from "./adapters";
+dotenv.config({ path: ".env" });
+
+const dummyLogger = winston.createLogger({
+  level: "debug",
+  transports: [new winston.transports.Console()],
+});
+
+const maxPriceAge = 300;
+
+class TestPriceClient extends PriceClient {
+  private static testInstance: TestPriceClient | undefined;
+
+  public static get(logger: Logger) {
+    if (!this.testInstance) this.testInstance = new TestPriceClient(logger);
+    return this.testInstance;
+  }
+
+  // Hack to return protected getPriceCache.
+  _getPriceCache(currency: string, platform: string): PriceCache {
+    return this.getPriceCache(currency, platform);
+  }
+
+  protected constructor(logger: Logger) {
+    super(logger);
+  }
+}
+
+function validateTokenPrice(tokenPrice: TokenPrice, address: string, timestamp: number) {
+  assert.ok(tokenPrice);
+  assert(tokenPrice.address === address);
+  assert(tokenPrice.price > 0);
+  assert(tokenPrice.timestamp > 0);
+  assert(
+    tokenPrice.timestamp > timestamp - maxPriceAge,
+    `Surprise timestamp received: ${tokenPrice.timestamp} (expected at least ${timestamp - maxPriceAge}).`
+  );
+}
+
+// this requires e2e testing, should only test manually for now
+describe("PriceClient", function () {
```

Comment:
Should add a test that verifies the token price is returned in the correct numerical format. For example, querying WETH against ETH, or USDC/DAI against USD terms. It should then be OK to confirm that the price returned is accurate +/- 3% or so.

Reply:
Done.